python/pyarrow/_compute.pyx

```diff
@@ -669,3 +669,13 @@ cdef class StrptimeOptions(FunctionOptions):
 
     cdef const CFunctionOptions* get_options(self) except NULL:
         return self.strptime_options.get()
+
+cdef class VarianceOptions(FunctionOptions):
+    cdef:
+        unique_ptr[CVarianceOptions] variance_options
```

Comment:
I'm not sure why `unique_ptr` is used in some of the other classes, but it shouldn't be necessary. See `MinMaxOptions` for an example without.

Reply:
In `PartitionNthOptions` we need it because it doesn't have a null constructor so it can't be allocated on the stack. That's not the case for `VarianceOptions`, though, so I'll take a look

Reply:
I've rewritten `VarianceOptions` without `unique_ptr`.

I think that with the other three options classes exposed in this PR (`SetLookupOptions`, `PartitionNthOptions` and `StrptimeOptions`) it's necessary to use `unique_ptr` or, alternatively, we would need to define default constructors